libd/src/d/parser/conditional.d

```diff
@@ -165,8 +166,9 @@ auto parseMixin(ItemType)(ref TokenRange trange) if(is(Mixin!ItemType)) {
 	
 	trange.match(TokenType.CloseParen);
 	location.spanTo(trange.front.location);
```

Comment:
it is correct. but for a strange reason.
while the semicolon after the `mixn()` is syntactically there to end the mixin-expression.
I use it to end whatever expression was mixed in 
`mixin("return 42");` -> `return  42;` 
the semicolon at the end of `42;` and the one that ends the mixin are the same.

I do agree that this is not as clean as it should be... The alternative is inserting a semicolon at the end of each mixed-in expression ... that would shift the nasty code from here to the expressiionVisitor... maybe this is the way to go.